packages/trader/src/Modules/Trading/Components/Elements/__tests__/purchase-fieldset.spec.tsx

```diff
@@ -0,0 +1,90 @@
+import React from 'react';
+import { render, screen } from '@testing-library/react';
+import { isDesktop, isMobile } from '@deriv/shared';
+import PurchaseFieldset from '../purchase-fieldset';
+import userEvent from '@testing-library/user-event';
+
+const default_mocked_props = {
+    basis: '',
+    buy_info: {},
+    currency: '',
+    growth_rate: 0.03,
+    has_cancellation: false,
+    index: 1,
+    info: {
+        has_error: false,
+        id: 'test_id',
+        message: 'test_message',
+        growth_rate: 0.03,
+        stake: '10',
+    },
+    is_accumulator: false,
+    is_disabled: false,
+    is_high_low: false,
+    is_loading: false,
+    is_market_closed: false,
+    is_multiplier: false,
+    is_proposal_empty: false,
+    is_proposal_error: false,
+    is_vanilla: false,
+    is_turbos: false,
+    onClickPurchase: jest.fn(),
+    onHoverPurchase: jest.fn(),
+    purchased_states_arr: [true, false],
+    setPurchaseState: jest.fn(),
+    type: '',
+};
+
+jest.mock('@deriv/shared', () => ({
+    ...jest.requireActual('@deriv/shared'),
+    isMobile: jest.fn(() => false),
+    isDesktop: jest.fn(() => true),
+}));
+jest.mock('Modules/Trading/Components/Form/Purchase/contract-info', () => jest.fn(() => <p>ContractInfo</p>));
+jest.mock('Modules/Trading/Components/Elements/purchase-button', () => jest.fn(() => <p>PurchaseButton</p>));
+jest.mock('../../Form/Purchase/cancel-deal-info', () => jest.fn(() => <p>CancelDealInfo</p>));
```

Comment:
@kate-deriv interesting choice, i wonder why p elements? :)

Reply:
Actually there is no secret souse) I didn't want to use inline tags like `<span>` , so was choosing between blocks mainly `<div>` and `<p>`. Can use `<div>` instead👍🏻

Reply:
@kate-deriv I would really appreciate if you could use divs please just to follow the common style of writing mocks 🙏🏻

Reply:
Will do